### PR TITLE
Make graph filename argument optional

### PR DESCRIPTION
**Pull request by JPXKQX**
This pull request makes the 'graph' argument of the GraphCreator class optional, allowing users to create graphs without saving them.

### CODE REVIEW

File: src/anemoi/graphs/create.py [hunk 1]
```diff
@@ -77,9 +77,14 @@ def clean(self, graph: HeteroData) -> HeteroData:
...
         """Save the graph to the output path."""
-        if not os.path.exists(self.path) or self.overwrite:
+        if self.path is None:
+            LOGGER.info("No output path specified. The graph will not be saved.")
+        elif not self.path.exists() or self.overwrite:
+            self.path.parent.mkdir(parents=True, exist_ok=True)
             torch.save(graph, self.path)
-            self.print(f"Graph saved at {self.path}.")
+            LOGGER.info(f"Graph saved at {self.path}.")
+        else:
```
Comment (theissenhelen) [inline src/anemoi/graphs/create.py:86]: This is somewhat duplicated from self.init(). It should probably be part of the same logic.

File: src/anemoi/graphs/create.py [hunk 2]
```diff
@@ -93,6 +98,9 @@ def _path_readable(self) -> bool:
         """Check if the output path is readable."""
         import torch
 
+        if self.path is None:
```
Comment (theissenhelen) [inline src/anemoi/graphs/create.py:101]: Should this be part of _path_readable()?

Comment (mchantry) [reply src/anemoi/graphs/create.py]: Where else would you put it?

Comment (theissenhelen) [reply src/anemoi/graphs/create.py]: If there is no path, you don't need to check whether the file is readable.

File: src/anemoi/graphs/create.py [hunk 3]
```diff
@@ -14,26 +18,9 @@ class GraphCreator:
 
     def __init__(
         self,
-        path,
-        config=None,
-        cache=None,
-        print=print,
-        overwrite=False,
-        **kwargs,
+        config: Union[PurePath, DotDict],
```
Comment (JesperDramsch) [inline src/anemoi/graphs/create.py:21]: nit: Thought about this more over the morning, I think this should actually be `Union[Path, DotDict]`.

**Review by JesperDramsch — APPROVED** [review]
LGTM, one nitpick, feel free to apply or ignore.

File: src/anemoi/graphs/create.py (hunk 3 above)
Comment (theissenhelen) [reply src/anemoi/graphs/create.py:21]: My thoughts exactly

**Review by mchantry — APPROVED** [review]
LGTM

**Pull request merged**